repair/repair_task.hh

```diff
@@ -0,0 +1,50 @@
+/*
+ * Copyright (C) 2022-present ScyllaDB
+ */
+
+/*
+ * SPDX-License-Identifier: AGPL-3.0-or-later
+ */
+
+#pragma once
+
+#include "task_manager/task_manager.hh"
+
+class repair_task : public task_manager::task::impl {
+public:
+    repair_task(task_manager::module_ptr module, task_id id, unsigned sequence_number = 0, std::string keyspace = "", std::string table = "", std::string type = "", std::string entity = "", task_manager::foreign_task_ptr&& parent = nullptr) 
+        : task_manager::task::impl(module, id, sequence_number, std::move(keyspace), std::move(table), std::move(type), std::move(entity), std::move(parent)) {
+        _status.progress_units = "SSTables";
+    }
+
+    future<task_manager::task::progress> get_progress() const override {
+        if (_children.empty()) {
+            return make_ready_future<task_manager::task::progress>(_progress);
+        }
+
+        return map_reduce(_children, [] (auto& child) {
+            return smp::submit_to(child.get_owner_shard(), [&child] {
+                return child->get_progress();
+            });
+        }, task_manager::task::progress{}, [] (task_manager::task::progress l, task_manager::task::progress r) {
+            task_manager::task::progress res{};
+            res.total = l.total + r.total;
+            res.completed = l.completed + r.completed;
+            return res;
+        });
+    }
+
+    ::is_abortable is_abortable() const noexcept override {
+        return ::is_abortable{_parent.get() == nullptr};
+    }
+
+    future<> abort() override {
+        // FIXME: implement
```

Comment:
I think that the cureent repair tasks are non abortable so there's nothing to fix.
That said, the abort_source should cause them to abort, but this is different from the abort() method that originates at the api layer.

Reply:
For using the abort source, the leaf task should subscribe to it and/or poll it to see if abort has been requested, and it should be used when blocking.